pylint/checkers/stdlib.py

```diff
@@ -332,6 +332,13 @@
 }
 
 
+DEPRECATED_ATTRIBUTES: DeprecationDict = {
+    (3, 12, 0): {
+        "calendar.January",
```

Comment:
You should add the other ones mentioned in the issue for 3.12. Someone who has time can go back and read the python release notes for earlier versions to see if we missed any in prior versions.